http/responseemitter.go

```diff
@@ -27,7 +27,7 @@ var (
 
 // NewResponseEmitter returns a new ResponseEmitter.
 func NewResponseEmitter(w http.ResponseWriter, method string, req *cmds.Request, opts ...ResponseEmitterOption) (ResponseEmitter, error) {
-	encType, enc, err := cmds.GetEncoder(req, w, cmds.JSON)
+	encType, enc, err := cmds.GetLocalEncoder(req, w, cmds.JSON)
```

Comment:
> What do you think about running DisplayCLI in the case that an HTTP client requests text/plain

In that case, we should continue to use the text encoder. `DisplayCLI` is specifically for controlling a terminal when a simple text encoder isn't sufficient.

We _currently_ use a `PostRun` function (which is run entirely client-side) but that ends up breaking `ipfs ... --enc=json` (because we intercept the response from the server before it hits the local encoder).